doc/select.md

```diff
@@ -0,0 +1,296 @@
+# Select examples
+
+## Filtering
+
+``CRUD`` allows to filter tuples by conditions. Each condition can use field name (or number) or index name. The first condition that uses index name is used to iterate over space. If there is no conditions that match index names, full scan is performed. Other conditions are used as additional filters. 
+
+**Note:** If you specify ``sharding key`` or ``bucket_id`` select will be performed on single node. Otherwise Map-Reduce over all nodes will be occurred.
+
+Below are examples of filtering data using these conditions. 
+
+### Getting space
+
+Let's check ``developers`` space contents to make other examples more clear. Just select first 6 values without conditions.
+
+**Example:**
+
+```lua
+crud.select('developers', nil, { first = 6 })
+---
+- metadata:
+  - {'name': 'id', 'type': 'unsigned'}
+  - {'name': 'bucked_id', 'type': 'unsigned'}
+  - {'name': 'name', 'type': 'string'}
+  - {'name': 'surname', 'type': 'string'}
+  - {'name': 'age', 'type': 'number'}
+  rows:
+  - [1, 7331, 'Alexey', 'Adams', 20]
+  - [2, 899, 'Sergey', 'Allred', 21]
+  - [3, 9661, 'Pavel', 'Adams', 27]
+  - [4, 501, 'Mikhail', 'Liston', 31]
+  - [5, 1993, 'Dmitry', 'Jacobi', 16]
+  - [6, 8765, 'Alexey', 'Sidorov', 51]
+...
+```
+
+### Select using index
+
+Let's say we have a ``age`` index. Example below gets a list of ``customers`` over 30 years old.
+
+**Example:**
+
+```lua
+crud.select('developers', {{'>=', 'age', 30}})
+---
+- metadata:
+  - {'name': 'id', 'type': 'unsigned'}
+  - {'name': 'bucked_id', 'type': 'unsigned'}
+  - {'name': 'name', 'type': 'string'}
+  - {'name': 'surname', 'type': 'string'}
+  - {'name': 'age', 'type': 'number'}
+  rows:
+  - [4, 501, 'Mikhail', 'Liston', 31]
+  - [6, 8765, 'Alexey', 'Sidorov', 51]
+...
+```
+
+### Select using composite index
+
+Suppose we have a composite index consisting of the ``name`` and ``surname`` fields. See example of select queries using such a composited index below.
+
+**Example**:
+
+```lua
+crud.select('developers', {{'==', 'full_name', {"Alexey", "Adams"}}})
+---
+- metadata:
+  - {'name': 'id', 'type': 'unsigned'}
+  - {'name': 'bucked_id', 'type': 'unsigned'}
+  - {'name': 'name', 'type': 'string'}
+  - {'name': 'surname', 'type': 'string'}
+  - {'name': 'age', 'type': 'number'}
+  rows:
+  - [1, 7331, 'Alexey', 'Adams', 20]
+...
+```
+
+### Select using partial key 
+
+Alternatively, you can use a partial key for a composite index.
```

Comment:
But if you specify `crud.select('developers', {{'==', 'full_name',  {nil, "Sidorov"}}})` than full scan by pk with filter by `full_name == {nil, "Sidorov"}` is performed.